graphql/schema/types/performer.graphql

```diff
@@ -0,0 +1,196 @@
+enum GenderEnum {
+  MALE
+  FEMALE
+  TRANSGENDER_MALE
+  TRANSGENDER_FEMALE
+  INTERSEX
+}
+
+enum FakeTitsEnum {
+  YES
+  NO
+  NA
+}
+
+type Measurements {
+  cup_size: String
+  # TODO - should these be in cm to be consistent with height?
+  bust: Int
+  waist: Int
+  hip: Int
+}
+
+input MeasurementsInput {
+  cup_size: String
+  # TODO - should these be in cm to be consistent with height?
+  bust: Int
+  waist: Int
+  hip: Int
+}
+
+type Performer {
+  id: ID!
+  name: String!
+  disambiguation: String
+  aliases: [String!]!
+  gender: GenderEnum
+  urls: [URL!]!
+  birthdate: Date
+  ethnicity: String
+  country: String
+  eye_color: String
+  """Height in cm"""
+  height: Int
+  measurements: Measurements!
+  # TODO - is there a better desciptor for this - boob_job? - enhanced?
+  fake_tits: FakeTitsEnum
+  career_start_year: Int
+  career_end_year: Int
+  tattoos: [String!]
+  piercings: [String!]
+}
+
+input PerformerCreateInput {
+  name: String!
+  disambiguation: String
+  aliases: [String!]
+  gender: GenderEnum
+  urls: [URLInput!]
+  birthdate: Date
+  ethnicity: String
+  country: String
+  eye_color: String
+  height: Int
+  measurements: MeasurementsInput
+  fake_tits: FakeTitsEnum
+  career_start_year: Int
+  career_end_year: Int
+  tattoos: [String!]
+  piercings: [String!]
+}
+
+input PerformerUpdateInput {
+  id: ID!
+  name: String
+  disambiguation: String
+  aliases: [String!]
+  gender: GenderEnum
+  urls: [URLInput!]
+  birthdate: Date
+  ethnicity: String
+  country: String
+  eye_color: String
+  height: Int
+  measurements: MeasurementsInput
+  fake_tits: FakeTitsEnum
+  career_start_year: Int
+  career_end_year: Int
+  tattoos: [String!]
+  piercings: [String!]
+}
+
+input PerformerDestroyInput {
+  id: ID!
+}
+
+input PerformerEditDetailsInput {
+  name: String
+  disambiguation: String
+  aliases: [String!]
+  gender: GenderEnum
+  urls: [URLInput!]
+  birthdate: Date
+  ethnicity: String
+  country: String
+  eye_color: String
+  height: Int
+  measurements: MeasurementsInput
+  fake_tits: FakeTitsEnum
+  career_start_year: Int
+  career_end_year: Int
+  tattoos: [String!]
+  piercings: [String!]
+}
+
+input PerformerEditInput {
+  """Not required for create type"""
+  id: ID
+  operation: OperationEnum!
+  """Not required for destroy type"""
+  details: PerformerEditDetailsInput
+  comment: String
+}
+
+type PerformerEdit {
+  name: String
+  disambiguation: String
+  added_aliases: [String!]
+  removed_aliases: [String!]
```

Comment:
I would argue that alias shouldn't be a fixed property on the performer, but rather a property of scene performances. So say a performer has the official name 'Mindy', but performed in a scene under the name 'Sugary M', then 'Sugary M' would be an alias attached to that performance, and would show up in the list of aliases for the performer 'Mindy'. There would be no need to add aliases since they would show up based on performance history. Of course, that would mean it wouldn't be possible to manually add an alias without a performance under that name, but I can't really think of a reason why you would want to do that.

Reply:
Adding a relationship between scene and alias is absolutely a good idea. I still like the idea of aliases for a performer however because there is bound to be cases where we add a performer before an exhaustive list of scenes can be populated. It would be useful to know a performer's aliases before the applicable scenes have been added. I'm thinking we could keep the alias list in the performer table, and just automatically add to the list when a performance with an alias is added.

Reply:
Sure, the only issue is that if someone assigns a scene to the wrong performer with an erroneous alias, and that gets fixed later, the erroneous alias would still persist on the performer. It might be useful to separate explicit aliases that have been set on the performer from scene credit aliases which are added implicitly, and just merge them together for the graphql query. That would also mean you could weight the aliases based on how many scenes it has been used in.